src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
```

Comment:
This method could be better this way:
`        return Arrays
                .stream(GetDivisors(candidate))
                .boxed()
                .collect(Collectors.toList())
                .stream()
                .limit(1000)
                .reduce(0L, Long::sum)
                .equals(candidate);`